flake-module.nix

```diff
@@ -0,0 +1,78 @@
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.treefmt = mkOption {
+          description = ''
+            treefmt flake module options
+          '';
+          type = types.submodule {
+            options = {
+              config = mkOption {
+                description = "treefmt-nix configuration";
+                type = types.submoduleWith {
+                  modules = (import ./.).all-modules pkgs;
+                };
+              };
+              module = mkOption {
+                type = types.raw; # TODO: module type?
+                description = "Evaluated module of treefmt-nix";
+                default =
+                  (import ./.).evalModule pkgs config.treefmt.config;
+              };
+              programs = mkOption {
+                type = types.attrsOf types.package;
+                description = "Attrset of formatter programs enabled in configuration";
+                default =
+                  pkgs.lib.concatMapAttrs
+                    (k: v:
+                      if v.enable
+                      then { "${k}" = v.package; }
+                      else { })
+                    config.treefmt.module.config.programs;
+              };
+              wrapper = mkOption {
+                type = types.package;
+                description = "Wrapper treefmt using the current configuration";
+                default = config.treefmt.module.config.build.wrapper;
+              };
+            };
+          };
+        };
+        config = {
+          checks.treefmt = pkgs.runCommandLocal "treefmt-check"
+            {
+              buildInputs = [ pkgs.git config.treefmt.wrapper ] ++ lib.attrValues config.treefmt.programs;
```

Comment:
Are you sure the programs are needed? I would assume that the treefmt config would already have a store reference to them, which is then referred to by the wrapper.

Reply:
True, oversight on my part (the old module [needed it](https://github.com/Platonic-Systems/treefmt-flake/blob/master/flake-module.nix#L48) because it had no wrapper). Fixed.